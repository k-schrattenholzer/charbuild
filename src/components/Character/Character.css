.Char-Container {
  width: 12em;
  padding: 1em;
  display:flex;
  align-items: center;
  padding: 10px;
  border-radius:  30px;
  box-shadow: var(--russian-green) 0px 4px 10px;
  margin-top: 1em;
  background-color: var(--laurel-green-2);
}