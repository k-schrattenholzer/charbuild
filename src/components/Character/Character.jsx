import React from 'react';
import './Character.css'

export default function Character({head, mid, bottom, shoes}) {
  return (
    <div 
    style={{display:'flex', flexDirection:'column'}}
    className="Char-Container"
    >

      <article
      className='head'
      aria-label='head'
      style={{
        backgroundImage: `url(./head/${head}.png)`,
        height: 150,
        width: 150,
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat',
      }}></article>

      <article
      className='mid'
      aria-label='mid'
      style={{
        backgroundImage: `url(./middle/${mid}.png)`,
        height: 150,
        width: 150,
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat',
      }}></article>

      <article
      className='bottom'
      aria-label='bottom'
      style={{
        backgroundImage: `url(./bottom/${bottom}.png)`,
        height: 150,
        width: 150,
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat',
      }}></article>

      <article
      className='shoes'
      aria-label='shoes'
      style={{
        backgroundImage: `url(./shoes/${shoes}.png)`,
        height: 150,
        width: 150,
        backgroundSize: 'cover',
        backgroundRepeat: 'no-repeat',
      }}></article>

    </div>
  )
}
