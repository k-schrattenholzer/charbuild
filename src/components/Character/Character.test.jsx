import { render, screen } from "@testing-library/react";
import Character from "./Character";

it('should display the Character component', () => {
  const {container}= render(<Character head='bat' mid='poncho' bottom='jeans' shoes='tennies'/>);

  const head = screen.getByLabelText(/head/);
  const top = screen.getByLabelText(/mid/);
  const bottom = screen.getByLabelText(/bottom/);
  const shoes = screen.getByLabelText(/shoes/);

  expect(head).toHaveStyle(`background-image: url(./head/bat.png)`);
  expect(top).toHaveStyle(`background-image: url(./middle/poncho.png)`);
  expect(bottom).toHaveStyle(`background-image: url(./bottom/jeans.png)`);
  expect(shoes).toHaveStyle(`background-image: url(./shoes/tennies.png)`);

  expect(container).toMatchSnapshot();
});
