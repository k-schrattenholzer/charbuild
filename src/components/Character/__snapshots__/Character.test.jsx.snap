// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the Character component 1`] = `
<div>
  <div
    class="Char-Container"
    style="display: flex; flex-direction: column;"
  >
    <article
      aria-label="head"
      class="head"
      style="background-image: url(./head/bat.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="mid"
      class="mid"
      style="background-image: url(./middle/poncho.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="bottom"
      class="bottom"
      style="background-image: url(./bottom/jeans.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      aria-label="shoes"
      class="shoes"
      style="background-image: url(./shoes/tennies.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
  </div>
</div>
`;
